crates/cast/src/starknet_commands/script/init.rs

```diff
@@ -0,0 +1,140 @@
+use anyhow::{anyhow, Context, Ok, Result};
+use camino::Utf8PathBuf;
+use std::fs;
+
+use cast::helpers::constants::SCRIPTS_DIR;
+use cast::helpers::response_structs::ScriptInitResponse;
+use cast::helpers::scarb_utils::get_cairo_version;
+use clap::Args;
+use indoc::{formatdoc, indoc};
+use scarb_metadata::ScarbCommand;
+
+#[derive(Args, Debug)]
+pub struct Init {
+    /// Name of a script to create
+    pub script_name: String,
+}
+
+pub fn init(init_args: &Init) -> Result<ScriptInitResponse> {
+    let script_root_dir_path = get_script_root_dir_path(&init_args.script_name)?;
+
+    init_scarb_project(&init_args.script_name, &script_root_dir_path)?;
+    add_dependencies(&script_root_dir_path)?;
+    modify_files_in_src_dir(&init_args.script_name, &script_root_dir_path)?;
+
+    Ok(ScriptInitResponse {
+        status: format!("Successfully initialized `{}`", init_args.script_name),
+    })
+}
+
+fn get_script_root_dir_path(script_name: &str) -> Result<Utf8PathBuf> {
+    let current_dir = std::env::current_dir()?;
+
+    let script_root_dir_path = current_dir
+        .file_name()
+        .and_then(|dir_name| dir_name.to_str())
+        .filter(|&dir_name| dir_name == SCRIPTS_DIR)
+        .map_or_else(
+            || current_dir.join(SCRIPTS_DIR).join(script_name),
```

Comment:
This looks counterintuitive to me

Reply:
> It's potentially useful when the user inits scripts project inside scarb package with contracts, but may be annoying when trying to init a scripts project in an arbitrary location.

that's true. I'm afraid that verifying this will add too much of a complexity - after all there might be some edge scenarios where we are (for example) in some workspace where do just want to have scripts, and the contracts are in differen workspace? (not sure, I'm just guessing here).

Reply:
On a second though - what do you guys think about such idea: we check if there is a `scripts` directory in cwd, if it is - we create a package dir in there, if it isn't we create pkg dir in cwd. After init we print out an info, something like "New package {xyz} created in {path}". So - we do not create scripts directory at all. This should simplify a few things, and we do not have this problem with removing empty scripts dir.

Reply:
From the user's perspective, this approach might be inconsistent. 
Whenever I use some CLI tool init command, I expect init to be executed in cwd.

Maybe a flag? Smth like `--path=<path-to-folder-where-script-project-will-be-initialized>`

Reply:
It also makes sense to me. What do you think @THenry14?

Reply:
I'd like to avoid adding a new flag to be honest. I think the behaviour where we create a package dir in cwd OR if there is scripts dir in cwd we create package dir in cwd/scripts makes the most sense here, as it's reasonable to expect we are in some workspace then. 

That would be probably also more consistent with cargo and scarb. It is reasonable to expect that tests are in tests directory, I guess the same should go for scripts. 

Of source, that would mean we should print out to the user the path where the script pkg was created. thoughts?

Reply:
hm, ok we can do that, but maybe let's impose stronger conditions that need to be met to create a script package in the `scripts` dir. Like the existence `Scarb.toml` file and `src` folder in cwd. WDYT?

Reply:
Going back to this:
> I think the behaviour where we create a package dir in cwd OR if there is scripts dir in cwd we create package dir in cwd/scripts makes the most sense here

now that I think about it, let's be stricter. let's require `scripts` directory, and drop the 'or create pkg dir in cwd'. This may become too complex to support, and as I said - it's not unreasonable to expect the scripts be in scripts directory. If someone wants to do it any other way, init command isn't for this person and it can always be made "manually"

> hm, ok we can do that, but maybe let's impose stronger conditions that need to be met to create a script package in the scripts dir. Like the existence Scarb.toml file and src folder in cwd. WDYT?

this may make sense. If toml and src are present, I'd suggest we just add a new script file + include it in mod. 
if they are not present (but the dir exists), we could probably just error out?
if the dir is not present, let's create it (along with toml, src, script file).

Does that make sense to you? or have I forgotten about something?

Reply:
> let's require scripts directory, and drop the 'or create pkg dir in cwd'.

Is there a plan to include the creation of a `scripts` folder in the `snforge init` command? If not, users will have to create this manually every time.

> If toml and src are present, I'd suggest we just add a new script file + include it in mod.

Should we do this only when we're in the `scripts` directory? Is there another way to verify that a Scarb project in the cwd is associated with a script? Generally I'm not sure that extending `script init` to add a new script file and include it in the `mod` is in the scope of `script init`.

Reply:
> Is there a plan to include the creation of a scripts folder in the snforge init command? If not, users will have to create this manually every time.

snforge has nothing to do with this, sncast should do this (in init subcommand)

> Should we do this only when we're in the scripts directory? Is there another way to verify that a Scarb project in the cwd is associated with a script? Generally I'm not sure that extending script init to add a new script file and include it in the mod is in the scope of script init.

this one is very tricky indeed. I think this _could_ be in scope, but it depends on us it it _will be_. If this it too complicated I'd just fall back to init just be able to create a simple layout (eg scripts directory with one example script inside, and nothing more)

Reply:
As we discussed previously, let's keep things straightforward:
1. The `script init` command should be used once per project to set up a template for a deployment script.
2. Always create a `scripts` directory and nest a script package directory within it. If there's already a `scripts` directory in the cwd, it should throw an error.

Please let me know if someone disagrees with this approach.

Reply:
I don't understand why we cannot run `init` multiple times in the same cwd with distinct script names?

Reply:
This increases the complexity of init unnecessarily. It ought to be a simple, straightforward way to create a script package with the directory layout etc, no need to try to cover all the possibilities. After all this is _init_, I don't think it's unreasonable to leave it up to the users to further modify the structure if needed (add more packages/scripts - of course we will need to improve our docs).

Further down the line, if there is a demand, we can think about expanding it. At the moment, in my opinion, it is sufficient as is.